Public/Src/Engine/UnitTests/Scheduler.IntegrationTest/OutputReorderTests.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Linq;
+using BuildXL.Pips.Operations;
+using BuildXL.Scheduler.Fingerprints;
+using BuildXL.Storage;
+using BuildXL.Utilities;
+using Test.BuildXL.Executables.TestProcess;
+using Test.BuildXL.Scheduler;
+using Test.BuildXL.TestUtilities.Xunit;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace IntegrationTest.BuildXL.Scheduler
+{
+    public class OutputReorderTests : SchedulerIntegrationTestBase
+    {
+        public OutputReorderTests(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Fact]
+        public void ReorderOutputsCacheHit()
+        {
+            FileArtifact outputFile1 = CreateSourceFile();
+            FileArtifact outputFile2 = CreateSourceFile();
+
+            var builderA = CreatePipBuilder(new Operation[]{});

```

Comment:
>Operation[]{}) [](start = 48, length = 14)

Can we not have a real write operation that write outputs to files?

Reply:
yes, I have that in my test, but it turns out to be doesn't affect the running the test, so I eliminate it to simplify the case. I will add it back.

Reply:
done